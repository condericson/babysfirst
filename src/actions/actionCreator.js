// add first
export function addMemory(userId, date, content, image) {
  console.log('dispatching add first');
  return {
    type: 'ADD_FIRST',
    userId,
    date,
    content,
    image,
  };
}

// // remove comment
// export function removeComment(postId, i) {
//   return {
//     type: 'REMOVE_COMMENT',
//     i,
//     postId
//   }
// }
