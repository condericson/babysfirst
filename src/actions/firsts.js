import { browserHistory } from 'react-router';
import axios from 'axios';
// import { Firsts } from '../utils/api';

export const GET_FIRSTS = 'GET_FIRSTS';
export const GET_FIRSTS_SUCCESS = 'GET_FIRSTS_SUCCESS';
export const GET_FIRSTS_ERROR = 'GET_FIRSTS_ERROR';

function getFirstsSuccess(firsts) {
  return {
    type: GET_FIRSTS_SUCCESS,
    firsts,
  };
}

function getFirstsError(error) {
  return {
    type: GET_FIRSTS_ERROR,
    error,
  };
}

export function getFirsts(userId) {
  return async dispatch => {
    dispatch({ type: GET_FIRSTS });
    try {
      const { data } = await axios.get(`/firsts/${userId}`);
      console.log('Data from axios get firsts', data);
      dispatch(getFirstsSuccess(data));
    } catch (e) {
      return dispatch(getFirstsError(e));
    }
    return browserHistory.push('/timeline');
  };
}

export const ADD_FIRST = 'ADD_FIRST';
export const ADD_FIRST_SUCCESS = 'ADD_FIRST_SUCCESS';
export const ADD_FIRST_ERROR = 'ADD_FIRST_ERROR';

function addFirstsSuccess(firsts) {
  return {
    type: ADD_FIRST_SUCCESS,
    firsts,
  };
}

function addFirstsError(error) {
  return {
    type: ADD_FIRST_ERROR,
    error,
  };
}

export function addFirsts(first) {
  return async dispatch => {
    dispatch({ type: ADD_FIRST });
    console.log('REACHED HERE');
    try {
      const { data } = await axios.post('/firsts', first);
      console.log('Data from axios first post', data);
      dispatch(addFirstsSuccess(data));
    } catch (e) {
      return dispatch(addFirstsError(e));
    }
    return browserHistory.push('/timeline');
  };
}
