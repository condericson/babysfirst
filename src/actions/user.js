import { browserHistory } from 'react-router';

import { User } from '../utils/api';

export const LOGIN = 'LOGIN';
export const LOGIN_SUCCESS = 'LOGIN_SUCCESS';
export const LOGIN_ERROR = 'LOGIN_ERROR';
export const LOGIN_INCORRECT_INFO = 'LOGIN_INCORRECT_INFO';

export function loginSuccess(user) {
  return {
    type: LOGIN_SUCCESS,
    user,
  };
}

export function loginError(error) {
  return {
    type: LOGIN_ERROR,
    error,
  };
}

export function loginIncorrectInfo(message) {
  return {
    type: LOGIN_INCORRECT_INFO,
    message,
  };
}

export function login(user) {
  return async dispatch => {
    dispatch({ type: LOGIN });
    try {
      const response = await User.login(user);
      if (response.response) {
        dispatch(loginIncorrectInfo(response.response.data.message));
      } else {
        dispatch(loginSuccess(response));
        return browserHistory.push('/timeline');
      }
    } catch (e) {
      return dispatch(loginError(e));
    }
  };
}

export const LOGOUT = 'LOGOUT';

export function logout() {
  return {
    type: LOGOUT,
  };
}

export const SIGNUP = 'SIGNUP';
export const SIGNUP_SUCCESS = 'SIGNUP_SUCCESS';
export const SIGNUP_ERROR = 'SIGNUP_ERROR';
export const SIGNUP_INCORRECT_INFO = 'SIGNUP_INCORRECT_INFO';

export function signupSuccess(user) {
  return {
    type: SIGNUP_SUCCESS,
    user,
  };
}

function signupError(error) {
  return {
    type: SIGNUP_ERROR,
    error,
  };
}

export function signupIncorrectInfo(message) {
  return {
    type: SIGNUP_INCORRECT_INFO,
    message,
  };
}

export function signup(user) {
  return async dispatch => {
    dispatch({ type: SIGNUP });
    try {
      const response = await User.signup(user);
      if (response.response) {
        dispatch(signupIncorrectInfo(response.response.data.message));
      } else {
        dispatch(signupSuccess(response));
        return browserHistory.push('/timeline');
      }
    } catch (e) {
      return dispatch(signupError(e));
    }
  };
}
