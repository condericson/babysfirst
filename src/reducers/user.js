import {
  LOGIN,
  LOGIN_SUCCESS,
  LOGIN_ERROR,
  LOGIN_INCORRECT_INFO,
  LOGOUT,
  SIGNUP,
  SIGNUP_ERROR,
  SIGNUP_SUCCESS,
  SIGNUP_INCORRECT_INFO,
} from '../actions/user';

const initialState = {
  loading: false,
  currentUserId: null,
  logged: false,
  error: null,
  signupError: null,
  loginError: null,
};

export default (state = initialState, action) => {
  switch (action.type) {
    case LOGIN:
      return {
        ...state,
        loading: true,
        loginError: null,
      };
    case LOGIN_SUCCESS:
      return {
        ...state,
        loading: false,
        currentUserId: action.user._id,
        currentUserBirthday: action.user.birthday,
        logged: true,
        signupError: null,
      };
    case LOGIN_ERROR:
      return {
        ...state,
        loading: false,
        loginError: action,
      };
    case LOGIN_INCORRECT_INFO:
      return {
        ...state,
        loading: false,
        loginError: action.message,
      };
    case LOGOUT:
      return initialState;
    case SIGNUP:
      return {
        ...state,
        loading: true,
        signupError: null,
      };
    case SIGNUP_SUCCESS:
      return {
        ...state,
        loading: false,
        currentUserId: action.user._id,
        currentUserBirthday: action.user.birthday,
        logged: true,
        signupError: null,
      };
    case SIGNUP_ERROR:
      return {
        ...state,
        loading: false,
        signupError: action.error,
      };
    case SIGNUP_INCORRECT_INFO:
      return {
        ...state,
        loading: false,
        signupError: action.message,
      };
    default:
      return state;
  }
};
