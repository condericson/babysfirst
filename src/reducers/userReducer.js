function recordUserId(state = [], action) {
  switch (action.type) {
    case 'RECORD_USER':
    // return the new state with the new comment
      return [...state, {
        currentUserId: action.userId,
      }];
    default:
      return state;
  }
}

export default recordUserId;
