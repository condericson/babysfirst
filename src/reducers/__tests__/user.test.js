import userReducer from '../user';
import { loginSuccess, signupSuccess, logout } from '../../actions/user';

const dummyData = {
  user1: [{
    _id: 12345,
    username: 'larry01',
    password: 'password01',
    birthday: '04-05-2016',
  }],
  user2: [{
    _id: 67890,
    username: 'bob02',
    password: 'password02',
    birthday: '05-06-2017',
  }],
  loading: false,
  errorMessage: false,
};

describe('#loginSuccess()', () => {
  it('should record the userId', () => {
    expect(
      userReducer(
        undefined,
        loginSuccess(
          dummyData.user1,
        )
      )
    ).toMatchSnapshot();
  });
});

describe('signupSuccess()', () => {
  it('should record the userId', () => {
    expect(
      userReducer(
        undefined,
        signupSuccess(
          dummyData.user1,
        )
      )
    ).toMatchSnapshot();
  });
});

describe('logout()', () => {
  it('should restore initialState', () => {
    expect(
      userReducer(
        undefined,
        logout()
      )
    ).toMatchSnapshot();
  });
});
