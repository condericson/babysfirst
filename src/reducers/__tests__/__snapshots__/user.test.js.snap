exports[`#loginSuccess() should record the userId 1`] = `
Object {
  "currentUserBirthday": undefined,
  "currentUserId": undefined,
  "error": null,
  "loading": false,
  "logged": true,
  "loginError": null,
  "signupError": null,
}
`;

exports[`logout() should restore initialState 1`] = `
Object {
  "currentUserId": null,
  "error": null,
  "loading": false,
  "logged": false,
  "loginError": null,
  "signupError": null,
}
`;

exports[`signupSuccess() should record the userId 1`] = `
Object {
  "currentUserBirthday": undefined,
  "currentUserId": undefined,
  "error": null,
  "loading": false,
  "logged": true,
  "loginError": null,
  "signupError": null,
}
`;
